code/modules/mining/mine_turf_types.dm

```diff
@@ -21,6 +22,37 @@
 		underlay_appearance.icon_state += "[rand(0,12)]"
 	return TRUE
 
+/turf/simulated/lava/Entered(atom/movable/AM, atom/oldloc)
+	if(istype(AM, /mob/living))
+		var/mob/living/L = AM
+		if(!istype(oldloc,/turf/simulated/lava))
+			to_chat(L, "<span class='warning'>You get drenched in magma from entering \the [src]!</span>")
+		if(ishuman(L))
+			if(L.lying)
+				L.visible_message("<span class='danger'>\The [L] is consumed by the \the [src]!</span>!")
+				L.dust()
+				return 1
+			else
+				L.adjustFireLoss(rand(20,50))
```

Comment:
Shouldn't deal fire damage directly, just let the body temp calcs injure the mob.